arcgis-ios-sdk-samples/Layers/Style WMS layers/StyleWebMapServiceLayerViewController.swift

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright © 2018 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import ArcGIS
+
+/// A view controller that manages the interface of the Style WMS Layers sample.
+class StyleWebMapServiceLayerViewController: UIViewController {
+    /// The map displayed in the map view.
+    let map: AGSMap
+    let layer: AGSWMSLayer
+    var styles = [String]()
+    
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView!
+    @IBOutlet weak var segmentedControl: UISegmentedControl!
+    
+    required init?(coder aDecoder: NSCoder) {
```

Comment:
Can we follow the same pattern as we do in other `UIViewControllers` where variables are declared as unconditionally unwrapped optionals and then handle all setting in `viewDidLoad`?

```swift
var map: AGSMap!
var layer: AGSWMSLayer!
```

I think what you did in overriding the `init` is totally fine but it doesn't match the pattern we have laid out in other View Controllers.

Reply:
I realize this deviates from the style adopted by other samples, but there are good reason to do so:

1. `viewDidLoad()` is not intended to be the initializer for the view controller. `init` is a more appropriate place to initialize members that are not dependent on the view being loaded.
2. The `AGSWMSLayer` needs to load before it can be displayed. By telling it to load in the initializer, we have increased the likelihood that it will have loaded by the time the map view is displayed to the user.

I strongly believe that this is a good model for new samples to follow moving forward.

Reply:
There are also benefits for the view controller's other properties:

1. They don't need to be `IUO`s because they are set in an initializer.
2. They can be immutable because they are never changed after initialization.

Reply:
Yes I agree with your reasoning. I also believe in code consistency, perhaps we adopt the pattern across all samples?